### PR TITLE
KOGITO-6962 DSL reorganization for flexibility

### PR DESCRIPTION
**Pull request by radtriste**
https://issues.redhat.com/browse/KOGITO-6962

This PR is a part of https://github.com/kiegroup/kogito-pipelines/pull/283 which is divided into smaller issues.
This will ease the review. 

Modifications:
- Job folders reorganized
  - Created Folder enum (with JobType and Enviroment)
    - Job generation in folder done if the folder is activated
    - DSL generation from the different repositories do no more have conditional on folders => easiest implementation
  - JobTypes: PULLREQUEST, NIGHTLY, RELEASE, TOOLS, OTHER
  - Isolated in different environments: DEFAULT, NATIVE, MANDREL, QUARKUS_MAIN, QUARKUS_BRANCH, SONARCLOUD and KOGITO_BDD
- Reorganized specific checks for nightly/pr
  - default
  - native
  - mandrel (old native LTS)
  - quarkus-main (splitted LTS into main/branch)
  - quarkus-branch (splitted LTS into main/branch)
- Added some JobUtils method like `createAllEnvsPerRepoPRJobs` which creates automatically PR jobs for all enabled environments.
- Moved `main/release/prepare-release-branch` to `0-prepare-release-branch` as it makes more sense to have it in the root folder
- Updated dsl `branch.yaml` config file to be able to disable environments for specific branch(es)
  - and added `productized_branch` configurable value
- Reviewed PR templates of dependent repos and updated with new PR comment values (default remains the same)
- Updated GH action for testing DSL. Now the script is more aware of owner/branch.
- Added pipelines downstream DSL checks so that we make sure changes in DSL does not break the tests on dependent repos

Related PRs:
- https://github.com/kiegroup/kogito-pipelines/pull/416
- https://github.com/kiegroup/drools/pull/4301
- https://github.com/kiegroup/kogito-runtimes/pull/2144
- https://github.com/kiegroup/optaplanner/pull/1935
- https://github.com/kiegroup/optaweb-vehicle-routing/pull/681
- https://github.com/kiegroup/optaweb-employee-rostering/pull/795
- https://github.com/kiegroup/optaplanner-quickstarts/pull/322
- https://github.com/kiegroup/kogito-apps/pull/1301
- https://github.com/kiegroup/kogito-examples/pull/1223
- https://github.com/kiegroup/kogito-images/pull/1225
- https://github.com/kiegroup/kogito-operator/pull/1171
- https://github.com/kiegroup/kie-tools/pull/941

### CODE REVIEW

**Comment by radtriste** [comment]
@Ginxo please have a look.
Note that there is a test commit currently which will be removed before the merge (hence the Draft status of this PR)

File: .ci/jenkins/Jenkinsfile.buildchain [hunk 1]
```diff
@@ -69,22 +73,24 @@ pipeline {
             }
             steps {
                 script {
-                    env.BUILD_MVN_OPTS_CURRENT = env.BUILD_MVN_OPTS_CURRENT ?: ''
-                    env.BUILD_MVN_OPTS_CURRENT += isSonarCloudAnalysis() ? ' -Prun-code-coverage' : ''
-
-                    def buildChainActionInfo = getBuildChainActionInfo()
+                    List opts_current = env.BUILD_MVN_OPTS_CURRENT ? [ env.BUILD_MVN_OPTS_CURRENT ] : []
```
Comment (Ginxo) [inline .ci/jenkins/Jenkinsfile.buildchain]: I would move this whole`BUILD_MVN_OPTS_CURRENT` thing to a method

**Review by Ginxo — APPROVED** [review]
I've checked specially the build-chain part and LGTM, just a recommendation. Many thanks

File: .ci/jenkins/Jenkinsfile.buildchain (hunk 1 above)
Comment (radtriste) [reply .ci/jenkins/Jenkinsfile.buildchain]: update

**Comment by radtriste** [comment]
jenkins retest this

**Comment by radtriste** [comment]
Set back as draft as I need to run some tests on some last changes

**Pull request merged**